File: src/dataset/python/beer-song_exercism-fruit-in/claude/ClaudeSonnet4_beer_song_v4.py
def recite(start, take=1):
    if take <= 0:
        return []
    
    result = []
    for i in range(take):
        n = start - i
        bottle_n = _bottle(n)
        bottle_n_lower = bottle_n.lower()
        next_n = (n + 99) % 100
        bottle_next_lower = _bottle(next_n).lower()
        take_down_text = _take_down(n)
        
        result.extend([
            f"{bottle_n} of beer on the wall, {bottle_n_lower} of beer.",
            f"{take_down_text}, {bottle_next_lower} of beer on the wall.",
            ""
        ])
    
    return result[:-1]


def verse(n):
    bottle_n = _bottle(n)
    bottle_n_lower = bottle_n.lower()
    next_n = (n + 99) % 100
    bottle_next_lower = _bottle(next_n).lower()
    take_down_text = _take_down(n)
    
    return [
        f"{bottle_n} of beer on the wall, {bottle_n_lower} of beer.",
        f"{take_down_text}, {bottle_next_lower} of beer on the wall.",
        ""
    ]


def take_down(n):
    return _take_down(n)


def bottle(n):
    return _bottle(n)


def _take_down(n):
    if n == 0:
        return "Go to the store and buy some more"
    elif n == 1:
        return "Take it down and pass it around"
    else:
        return "Take one down and pass it around"


def _bottle(n):
    if n == 0:
        return "No more bottles"
    elif n == 1:
        return "1 bottle"
    else:
        return f"{n} bottles"

File: src/dataset/c/beer-song_exercism-IrvinRamiA/claude/ClaudeSonnet4_beer_song_v4.c
#include <stdlib.h>
#include <stdio.h>
#include <string.h>
#include "beer_song.h"

void recite(uint8_t start_bottles, uint8_t take_down, char **song)
{
    static const char* const templates[] = {
        "No more bottles of beer on the wall, no more bottles of beer.",
        "Go to the store and buy some more, 99 bottles of beer on the wall.",
        "%d bottle of beer on the wall, %d bottle of beer.",
        "Take it down and pass it around, no more bottles of beer on the wall.",
        "%d bottles of beer on the wall, %d bottles of beer.",
        "Take one down and pass it around, %d bottle of beer on the wall.",
        "Take one down and pass it around, %d bottles of beer on the wall."
    };
    
    uint16_t song_idx = 0;
    uint8_t bottles = start_bottles;
    
    for (uint8_t verse = 0; verse < take_down; verse++) {
        if (bottles == 0) {
            strcpy(song[song_idx++], templates[0]);
            strcpy(song[song_idx++], templates[1]);
        } else if (bottles == 1) {
            sprintf(song[song_idx++], templates[2], bottles, bottles);
            strcpy(song[song_idx++], templates[3]);
        } else if (bottles == 2) {
            sprintf(song[song_idx++], templates[4], bottles, bottles);
            sprintf(song[song_idx++], templates[5], bottles - 1);
        } else {
            sprintf(song[song_idx++], templates[4], bottles, bottles);
            sprintf(song[song_idx++], templates[6], bottles - 1);
        }
        
        if (verse < take_down - 1) {
            song[song_idx++][0] = '\0';
        }
        
        bottles = (bottles == 0) ? 99 : bottles - 1;
    }
}